src/components/Editor/Tabs.js

```diff
@@ -21,24 +21,36 @@ const { formatKeyShortcut } = require("../../utils/text");
 require("./Tabs.css");
 
 /*
- * Finds the hidden tabs by comparing the tabs' top offset.
- * hidden tabs will have a great top offset.
+ * Finds the hidden tabs by comparing the tabs' left offset with the right offset of parent.
+ * hidden tabs will have a greater left offset. (dir="ltr")
  *
  * @param sourceTabs Immutable.list
  * @param sourceTabEls HTMLCollection
  *
  * @returns Immutable.list
  */
-function getHiddenTabs(sourceTabs, sourceTabEls) {
+function getHiddenTabs(sourceTabsParent, sourceTabs, sourceTabEls) {
   sourceTabEls = [].slice.call(sourceTabEls);
-  function getTopOffset() {
-    const topOffsets = sourceTabEls.map(t => t.getBoundingClientRect().top);
-    return Math.min(...topOffsets);
+  
+  const sourceTabMinWidth = sourceTabEls.length ? 
+    window.getComputedStyle(sourceTabEls[0]).getPropertyValue("min-width") :
+    0;
+  const sourceTabDirection = sourceTabEls.length ? 
+    window.getComputedStyle(sourceTabEls[0]).getPropertyValue("direction") :
+    "ltr";
+  const parentOffset = sourceTabDirection === "ltr" ?
+    sourceTabsParent.getBoundingClientRect().right :
+    sourceTabsParent.getBoundingClientRect().left;
+
+  if (sourceTabDirection === "ltr") {
```

Comment:
also maybe add a `getTabOffset` helper

```js
function getTabOffset(tab, dir) {
  return tab.getBoundingClientRect()[dir]
}
```

this will abstract out the logic a bit